server/src/setup/setup.ts

```diff
@@ -0,0 +1,72 @@
+import multipart from '@fastify/multipart';
+
+import { sessionRoutes } from '~/domain/session/session.routes';
+import { postRoutes } from '~/domain/example/post/post.routes';
+import { config } from '~/utils/config';
+import { log } from '~/utils/log';
+import { authPlugin } from './auth';
+import { contextPlugin } from './context';
+import { type ServerInstance } from './server';
+import { setupErrorHandler } from './error';
+import { csrfPlugin } from './csrf';
+import { setupGraphQL } from './graphql/server';
+import { infraRoutes } from './infra.routes';
+
+export async function setupServer(server: ServerInstance) {
+  server.register(multipart);
+
+  /**
+   * Include context (db, log, user, session, etc.) to the request lifecycle.
+   * Accessible as `request.ctx` from route handlers.
+   */
+  await server.register(contextPlugin);
+
+  /**
+   * Cross-Site Request Forgery (CSRF) protection for production environments.
+   * See: https://lucia-auth.com/guides/validate-session-cookies/
+   */
+  await server.register(csrfPlugin, {
+    enabled: config.NODE_ENV === 'production',
+  });
+
+  /**
+   * Parse and validate the session from cookies and expose `fastify.authenticate`
+   * decorator which can be used to protect REST endpoints.
+   */
+  await server.register(authPlugin);
+
+  /**
+   * Setup error handler to catch and log errors.
+   * Includes Sentry integration to capture exceptions.
+   */
+  setupErrorHandler(server);
+
+  await setupGraphQL(server);
+
+  await server.register(infraRoutes); // health checks, etc.
+
+  // NOTE: if you are using GraphQL for all your API endpoints, you can remove these:
```

Comment:
We should document this clearly.

To make it easier for developers to identify the parts of the code that need to be edited when starting a new project, I suggest we adopt a consistent keyword in the beginning of the comment, such as `[NEW PROJECT INSTRUCTIONS]` or something similar. This will allow to quickly search and locate all relevant sections for updates.

This convention should be mentioned in the documentation so that it’s clear and easily discoverable for anyone working on the project.

Reply:
Good idea 👍🏻